subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -460,6 +480,88 @@ uint8_t ull_scan_disable(uint8_t handle, struct ll_scan_set *scan)
 	return 0;
 }
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+void ull_scan_done(struct node_rx_event_done *done)
+{
+	struct node_rx_hdr *rx_hdr;
+	struct ll_scan_set *scan;
+	struct lll_scan *lll;
+	uint8_t handle;
+	uint32_t ret;
+
+	lll = (void *)HDR_ULL2LLL(done->param);
+	scan = (void *)HDR_LLL2EVT(lll);
+	if (likely(scan->duration_lazy || !lll->duration_reload ||
+		   lll->duration_expire)) {
+		return;
+	}
+
+	rx_hdr = (void *)scan->node_rx_scan_term;
+	if (!rx_hdr) {
+		return;
+	}
+
+	handle = ull_scan_handle_get(scan);
+	LL_ASSERT(handle < BT_CTLR_SCAN_SET);
+
+#if defined(CONFIG_BT_CTLR_PHY_CODED)
+	if (handle == SCAN_HANDLE_1M) {
+		struct ll_scan_set *scan_coded;
+
+		scan_coded = ull_scan_set_get(SCAN_HANDLE_PHY_CODED);
+		scan_coded->node_rx_scan_term = NULL;
+	} else {
+		struct ll_scan_set *scan_1m;
+
+		scan_1m = ull_scan_set_get(SCAN_HANDLE_1M);
+		scan_1m->node_rx_scan_term = NULL;
+	}
+#endif /* CONFIG_BT_CTLR_PHY_CODED */
+
+	rx_hdr->type = NODE_RX_TYPE_EXT_SCAN_TERMINATE;
+	rx_hdr->handle = handle;
+
+	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_HIGH,
+			  (TICKER_ID_SCAN_BASE + handle), ticker_op_ext_stop_cb,
+			  scan);
+
+	LL_ASSERT((ret == TICKER_STATUS_SUCCESS) ||
+		  (ret == TICKER_STATUS_BUSY));
+}
+
+void ull_scan_term_dequeue(uint8_t handle)
+{
+	struct ll_scan_set *scan;
+
+	scan = ull_scan_set_get(handle);
+	scan->is_enabled = 0U;
```

Comment:
Any risk that `ull_scan_set_get` returns NULL here?

Reply:
Seems like there are plenty of other places where there is a NULL check following this get.

Reply:
I have only added the check in one place, other places the caller uses a constant valid handle value or `SCAN_HANDLE_1M` or `SCAN_HANDLE_PHY_CODED`